### DISCUSSION TITLE
无限字符滚动横幅拼字(991cnx)

### DISCUSSION THREAD

**Discussion by qiufuyu123** [Physics365/991CN-X-CW-Decompilation#32] [Show and tell]
https://github.com/Physics365/991CN-X-CW-Decompilation/assets/60765480/169daf2c-9ab8-4bef-a3a7-03636f06c90a

参考ROP:  
[ROP](https://github.com/Physics365/991CN-X-CW-Decompilation/blob/main/ROP/ROPCodes/Examples/RollingPrint.asm)

**Comment by Physics365** [comment]
强👍👍👍

**Comment by fish4terrisa-MSDSM** [comment]
好厉害的说！不过呐，有没有除了那个F.xxxxxxx -> x以外的刷字方法？那个好像会导致执行完后按开机键初始化（因为第一个字符不合法）

**Comment by EwTE-H** [comment]
打长段文字可以用我那个内存编辑器，因为显示的都是二级字符所以打错了很容易发现。另外编辑器本身也有显示的功能

**Reply by fish4terrisa-MSDSM** [reply]
那大佬您可以把它上传一下repo吗，感觉还蛮常用的

**Reply by EwTE-H** [reply]
直接全文复制原帖好了
我这边网络环境不好，上传文件老要挂，可能还得要各位帮个忙

**Reply by fish4terrisa-MSDSM** [reply]
大佬能说明一下怎么用吗，毕竟好像和见过的其他rop不太一样，没有bin文件，也没有说明怎么运行，能添加一下说明吗，看着那个贴吧有点茫然的说(不知道怎么打)

**Reply by EwTE-H** [reply]
照着一楼的图打对应字符就行，第一行的D2D2是开始编辑的地方，用的时候替换成需要的值，打完按=即可，和以前拼字等“传统”rop一样
没有bin是因为，这个程序比第一个有bin的rop（QQR3MMP的贪吃蛇）早一天发布，而且这个程序长度不到200字节，可以直接在输入区打出来，不需要通过quickcpy啥的
